File: annotation-processor-testing/src/main/java/com/tickaroo/tikxml/annotationprocessing/propertyelement/EmptyIntPropertyElement.java
package com.tickaroo.tikxml.annotationprocessing.propertyelement;

import com.tickaroo.tikxml.annotation.PropertyElement;
import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "emptyPropertyTag")
public class EmptyIntPropertyElement {
  @PropertyElement
  int empty;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof EmptyIntPropertyElement)) return false;

    EmptyIntPropertyElement that = (EmptyIntPropertyElement) o;

    return empty == that.empty;
  }

  @Override public int hashCode() {
    return empty;
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/propertyelement/EmptyIntPropertyElement.kt
package com.tickaroo.tikxml.annotationprocessing.propertyelement

import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "emptyPropertyTag")
class EmptyIntPropertyElement {
    @PropertyElement
    var empty: Int = 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is EmptyIntPropertyElement) return false

        val that = other as EmptyIntPropertyElement?

        return empty == that!!.empty
    }

    override fun hashCode(): Int {
        return empty
    }
}
